components/hab/src/command/pkg/binlink.rs

```diff
@@ -208,16 +197,65 @@ impl Binlink {
         Err(Error::CannotParseBinlinkSource(path.as_ref().to_path_buf()))
     }
 
-    pub fn link(&self) -> Result<()> {
-        let template = format!("@echo off\nREM source='{0}'\n\"{0}\" %*",
-                               self.src.display());
-        fs::write(&self.dest, template)?;
+    pub fn link(&self, env: HashMap<String, String>) -> Result<()> {
+        self.write_binstub(self.stub_template(env).as_bytes())
+    }
+
+    #[cfg(not(target_os = "windows"))]
+    fn stub_template(&self, env: HashMap<String, String>) -> String {
+        let mut exports = String::new();
+        for (key, value) in env.into_iter() {
+            exports.push_str(&format!("export {}={}\n", key, value));
+        }
+        format!(
+                "\
+#!/bin/sh
+# source='{0}'
+{1}
+{0} \"$@\"",
```

Comment:
You might want to consider using [exec](https://en.wikipedia.org/wiki/Exec_(system_call)) to turn the shell script process into the target program (`exec {0} \"$@\""`)

Reply:
If you use Rust's [raw string literals](https://doc.rust-lang.org/reference/tokens.html#raw-string-literals) here, then you can avoid needing to escape the double quotes (it's an odd looking string like `r#"I can use "double quotes""#`). The only catch is that the `"\` on line 211 would have to become `"#!/bin/sh` as the backslash would be added into the string.